src/libraries/Native/Unix/System.Native/pal_memory.c

```diff
@@ -1,27 +1,54 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+#include "pal_config.h"
 #include "pal_memory.h"
 
 #include <stdlib.h>
 #include <string.h>
 
-void* SystemNative_MemAlloc(uintptr_t size)
+void* SystemNative_AlignedAlloc(uintptr_t alignment, uintptr_t size)
 {
-    return malloc(size);
+#if HAVE_ALIGNED_ALLOC && !defined(__APPLE__)
+    // We want to prefer the standardized aligned_alloc function. However
+    // it cannot be used on __APPLE__ since we target 10.13 and it was
+    // only added in 10.15, but we might be compiling on a 10.15 box.
```

Comment:
We build on 10.15 for bits intended to run on 10.13?

Reply:
We build on 10.15 saying we target 10.13 and the headers have annotations saying "this API is only available on 10.xx+"

Similarly to say building for 10.0.10240 (Threshold 1) on a machine running 19043 (21H1).

Reply:
> the headers have annotations saying "this API is only available on 10.xx+" so using them raises a warning

And that doesn't cause check_symbol_exists to fail?  Is there an alternative configure.cmake construct we could use, e.g. check_c_source_compiles, that would enable HAVE_ALIGNED_ALLOC to be defined "correctly"?  We're almost certainly going to forget about this !defined(__APPLE__) even after we're only targeting versions that have the API available.

Reply:
There is probably some other CMake magic or define checks that could be done here, but I think that can be tracked via a separate issue for follow up.

This impacts a range of Apple products (iOS, M1, OSX, etc) and may not be trivial to do correctly on all platforms.

Reply:
As long as it actually is tracked somewhere for us to follow-up on.

Reply:
https://github.com/dotnet/runtime/issues/54296

Reply:
Here is the existing place where all similar adjustment are concentrated today:

https://github.com/dotnet/runtime/blob/9d771a26f058a9fa4a49850d4778bbab7aa79a22/src/libraries/Native/Unix/configure.cmake#L532-L558

We may want to move this one to this central place too.

Reply:
Fixed.